gradle/plugins/src/main/kotlin/com/saveourtool/save/buildutils/PublishingConfiguration.kt

```diff
@@ -65,11 +78,25 @@ internal fun Project.configurePublications() {
     }
 }
 
-internal fun Project.configureSigning() {
-    configure<SigningExtension> {
-        useInMemoryPgpKeys(property("signingKey") as String?, property("signingPassword") as String?)
-        logger.lifecycle("The following publications are getting signed: ${extensions.getByType<PublishingExtension>().publications.map { it.name }}")
-        sign(*extensions.getByType<PublishingExtension>().publications.toTypedArray())
+/**
+ * Enables signing of the artifacts if the `signingKey` project property is set.
+ *
+ * Should be explicitly called after each custom `publishing {}` section.
+ */
+fun Project.configureSigning() {
+    if (hasProperty("signingKey")) {
+        configure<SigningExtension> {
+            useInMemoryPgpKeys(property("signingKey") as String?, property("signingPassword") as String?)
+            val publications = extensions.getByType<PublishingExtension>().publications
+            val publicationCount = publications.size
+            val message = "The following $publicationCount publication(s) are getting signed: ${publications.map(Named::getName)}"
+            val style = when (publicationCount) {
+                0 -> Failure
+                else -> Success
+            }
+            styledOut(logCategory = "signing").style(style).println(message)
```

Comment:
does `println` much better than `logger`?

Reply:
It's a styled (colored) output. 

The message is important, and I'd prefer it to stand out of the rest of the output.

Reply:
discussed offline, logger doesn't fit here, so created custom println here